test/unit/alignment/pairwise/edit_distance_unbanded_test_template.hpp

```diff
@@ -0,0 +1,148 @@
+#pragma once
+
+#include <gtest/gtest.h>
+
+#include <seqan3/alignment/matrix/alignment_score_matrix.hpp>
+#include <seqan3/alignment/matrix/alignment_trace_matrix.hpp>
+#include <seqan3/alignment/pairwise/edit_distance_unbanded.hpp>
+
+#include <seqan3/range/view/to_char.hpp>
+
+#include "fixture/alignment_fixture.hpp"
+
+using namespace seqan3;
+using namespace seqan3::detail;
+using namespace seqan3::test::alignment::fixture;
+
+template <typename word_t = uint64_t, typename is_semi_global_t = std::false_type>
+struct test_traits_type
+{
+    using word_type = word_t;
+    using is_semi_global_type = is_semi_global_t;
+};
+
+template <auto _fixture, typename word_t>
+struct global_fixture : public ::testing::Test
+{
+    auto fixture() -> decltype(alignment_fixture{*_fixture}) const &
```

Comment:
Any specific reason for using trailing return type? (I don't mind that much, but AFAIK we are not using it anywhere (outside concepts?))

Reply:
This is a left-over of an old idea.

You can write something like this to emphasise that you know the return type
```c++
std::vector result = computation(); // I know it is a vector over some elements
```

But you can't omit the template  when defining a function

```c++
std::vector computation() // does not work
{}

std::vector<auto> computation() // also does not work
{}
```

but with the following you can communicate the return type

```c++
auto computation() -> decltype(std::vector{tmp})
{
   // ...
   return tmp;
}
```

At some point we said we wanted to make it more clear what the return types are, but I think this idea never really gained momentum.